sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -3491,17 +3521,41 @@ case class ArrayDistinct(child: Expression)
             body
           }
 
-        val processArray = withArrayNullAssignment(
+        def withNaNCheck(body: String): String = {
```

Comment:
shall we move these codegen utils to `SQLOpenHashSet` as well to reduce duplicated code?

Reply:
> shall we move these codegen utils to `SQLOpenHashSet` as well to reduce duplicated code?

How about to do this after all done.

Reply:
@cloud-fan Updated, How about current?